packages/application-generic/src/services/schedulers/providers/agenda.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  Agenda,
+  IDatabaseOptions,
+  IDbConfig,
+  IMongoOptions,
+} from '@hokify/agenda';
+
+export type AgendaFactoryOptions = {
+  name?: string;
+  defaultConcurrency?: number;
+  processEvery?: string | number;
+  maxConcurrency?: number;
+  defaultLockLimit?: number;
+  lockLimit?: number;
+  defaultLockLifetime?: number;
+} & (IDatabaseOptions | IMongoOptions) &
+  IDbConfig;
+
+const getAgendaOpts = (): AgendaFactoryOptions => {
+  return { db: { address: String(process.env.MONGO_URL) } };
```

Comment:
should we specify custom collection name? if yes, does it make sense to make it an env variable as it doesnt change often.
default collection name: `agendaJobs`

Reply:
it creates automatically, no migrations needed.